### PR TITLE
Added configurable timeouts for fetching synthetics data

### PR DESCRIPTION
**Pull request by lukaa12**
KNTK-254

### CODE REVIEW

File: data/config.yaml [hunk 1]
```diff
@@ -7,6 +7,9 @@ data_update_period_seconds: 60
 # time window to fetch the test results for
 data_update_lookback_seconds: 300
 
+# timeout for fetching synthetics tests results
```
Comment (mateuszmidor) [inline data/config.yaml:10]: please explain in comment why 2 numbers

File: domain/cached_repo_request_driven.py [hunk 2]
```diff
@@ -24,6 +25,7 @@ def __init__(
         monitored_test_id: TestID,
         max_data_age_seconds: int,
         lookback_seconds: int,
+        timeout: Optional[Tuple[float, float]] = None,
```
Comment (mateuszmidor) [inline domain/cached_repo_request_driven.py:28]: I think it would be nice to have a default timeout of eg. 30 seconds

File: domain/repo.py [hunk 3]
```diff
@@ -7,5 +7,7 @@
 class Repo(Protocol):
     """ Repo provides data access to Kentik Synthetic Tests """
 
-    def get_mesh_test_results(self, test_id: TestID, results_lookback_minutes: int) -> MeshResults:
+    def get_mesh_test_results(
+        self, test_id: TestID, results_lookback_minutes: int, timeout: Optional[Tuple[float, float]] = None
```
Comment (mateuszmidor) [inline domain/repo.py:11]: Default timeout would be nice

File: main.py [hunk 4]
```diff
@@ -16,7 +16,7 @@
 from presentation.matrix_view import make_mesh_test_matrix_layout
 
 FORMAT = "[%(asctime)-15s] [%(process)d] [%(levelname)s]  %(message)s"
-logging.basicConfig(level=logging.INFO, format=FORMAT)
+logging.basicConfig(level=logging.DEBUG, format=FORMAT)
```
Comment (mateuszmidor) [inline main.py:19]: As per comment in earlier PR, logging level should stay INFO

File: domain/repo.py (hunk 3 above)
Comment (mateuszmidor) [reply domain/repo.py:11]: Maybe timeout should be configured in Repo initialization? I think it wont change from call to call

File: infrastructure/data_access/http/synthetics_repo.py [hunk 5]
```diff
@@ -24,13 +24,15 @@ class SyntheticsRepo:
     def __init__(self, email, token: str) -> None:
         self._api_client = KentikAPI(email=email, token=token)
 
-    def get_mesh_test_results(self, test_id: TestID, results_lookback_seconds: int) -> MeshResults:
+    def get_mesh_test_results(
+        self, test_id: TestID, results_lookback_seconds: int, timeout: Optional[Tuple[float, float]] = None
```
Comment (mateuszmidor) [inline infrastructure/data_access/http/synthetics_repo.py:28]: Please move timeout to __init__ function

**Review by mateuszmidor — APPROVED** [review]

File: data/config.yaml (hunk 1 above)
Comment (lukaa12) [reply data/config.yaml:10]: Added appropriate comment line below

File: domain/repo.py (hunk 3 above)
Comment (lukaa12) [reply domain/repo.py]: Agree, moved setting timeout to init func

**Review by Fenthick — APPROVED** [review]

File: domain/cached_repo_request_driven.py [hunk 6]
```diff
@@ -48,13 +51,13 @@ def _update_cache_if_needed(self) -> None:
 
         logger.debug("Updating data cache start...")
         try:
-            results = self._source_repo.get_mesh_test_results(self._test_id, self._lookback_seconds)
+            results = self._source_repo.get_mesh_test_results(self._test_id, self._lookback_seconds, self._timeout)
             with self._cache_access_lock:
                 self._cache_test_results = results
                 self._cache_timestamp = datetime.now()
```
Comment (mmac-m3a) [inline domain/cached_repo_request_driven.py:55]: I recommend to use UTC time because it does not have daylight saving and hence avoids problems with interval computation during switchover.
... and as I've mentioned in review to the other pull request, the timestamp should be in the `MeshResults` class.

File: main.py (hunk 4 above)
Comment (mmac-m3a) [reply main.py]: Log level also needs to be configurable via config file and ideally also an environment variable (which are easier to pass to Docker containers as a transient config).

File: domain/cached_repo_request_driven.py (hunk 6 above)
Comment (lukaa12) [reply domain/cached_repo_request_driven.py:55]: I agree. I will be better to do it in separate PR

File: data/config.yaml [hunk 7]
```diff
@@ -11,9 +11,12 @@ data_update_lookback_seconds: 300
 # numbers are specifying (connection, read) timeouts in seconds. Defaults are  30 secs for both
 timeout: [10.0, 10.0]
 
+# sets the logging level ppossible values are: [CRITICAL, ERROR, WARNING, INFO, DEBUG]
+logging_level: INFO
+
 # metric thresholds
 thresholds:
-  latency: # thresholds in microseconds
+  latency: # thresholds in microsecondslogging.setLevel
```
Comment (mateuszmidor) [inline data/config.yaml:19]: Is this change intended?

Comment (lukaa12) [reply data/config.yaml:19]: Nope :)

**Review by mateuszmidor — APPROVED** [review]

File: infrastructure/config/config_yaml.py [hunk 8]
```diff
@@ -49,5 +55,23 @@ def __init__(self, filename: str) -> None:
...
+            if config["logging_level"] == "ERROR":
+                return logging.ERROR
+            if config["logging_level"] == "WARN" or config["logging_level"] == "WARNING":
+                return logging.WARNING
+            if config["logging_level"] == "INFO":
+                return logging.INFO
+            if config["logging_level"] == "DEBUG":
+                return logging.DEBUG
+            else:
+                return logging.NOTSET
```
Comment (Fenthick) [inline infrastructure/config/config_yaml.py:75]: I think that having typo in config should lead to throwing error, because logging is important part of app.

File: infrastructure/config/config_yaml.py [hunk 9]
```diff
@@ -37,6 +39,10 @@ def packet_loss(self) -> Thresholds:
     def timeout(self) -> Tuple[float, float]:
         return self._timeout  # type: ignore
 
+    @property
+    def logging_level(self) -> int:
+        return self._logging
```
Comment (Fenthick) [inline infrastructure/config/config_yaml.py:44]:
```suggestion
        return self._logging_level
```
For simplicity

File: main.py [hunk 10]
```diff
@@ -15,8 +15,14 @@
 from presentation.main_view import make_page_layout
 from presentation.matrix_view import make_mesh_test_matrix_layout
 
+try:
```
Comment (Fenthick) [inline main.py:18]: I think that it belongs to WebApp.__init__()

**Review by Fenthick — CHANGES_REQUESTED** [review]

**Review by Fenthick — APPROVED** [review]

File: data/config.yaml [hunk 11]
```diff
@@ -7,6 +7,13 @@ data_update_period_seconds: 60
 # time window to fetch the test results for
 data_update_lookback_seconds: 300
 
+# timeout for fetching synthetics tests results
+# numbers are specifying (connection, read) timeouts in seconds. Defaults are  30 secs for both
```
Comment (mmac-m3a) [inline data/config.yaml:11]: Why not make it easier for users and have the timeouts separate in the config?
```
timeouts:
   read: 10
   connection: 10
```

**Review by mmac-m3a — APPROVED** [review]
